script.js

```diff
@@ -0,0 +1,128 @@
+// Task 1. Polyfills
+
+// Task 1.1
+console.log("Task 1.1");
+
+const notes = [
+  {
+    id: 1,
+    title: "Recipe",
+    description: "Ingredients include 2 eggs...",
+    pagesCount: 2,
+    isMarked: false,
+    access: [],
+  },
+  {
+    id: 2,
+    title: "Workouts",
+    description: "3 sets of squats...",
+    pagesCount: 1,
+    isMarked: true,
+    access: [],
+  },
+  {
+    id: 3,
+    title: "Passwords",
+    description: "VISA ...",
+    pagesCount: 6,
+    isMarked: true,
+    access: [],
+  },
+  {
+    id: 4,
+    title: "To Do 2021",
+    description: "1. Learn JS...",
+    pagesCount: 3,
+    isMarked: false,
+    access: [],
+  },
+];
+
+Array.prototype.map = function (callback) {
+  const newArray = [];
+  for (let i = 0; i < this.length; i++) {
+    newArray.push(callback(this[i], i, this));
+  }
+  return newArray;
+};
+
+const notesView = notes.map((note) => {
+  return { id: note.id, title: note.title };
+});
+console.log(notesView);
+
+// Task 1.2
+console.log("Task 1.2");
+
+Array.prototype.filter = function (callback) {
+  const newArray = [];
+  for (let i = 0; i < this.length; i++) {
+    if (callback(this[i], i, this)) {
+      newArray.push(this[i]);
+    }
+  }
+  return newArray;
+};
+
+const markedNotes = notes.filter((note) => note.isMarked === true);
+console.log(markedNotes);
+
+// Task 1.3
+console.log("Task 1.3");
+
+Array.prototype.reduce = function (callback, initialValue) {
+  let accumulator = initialValue !== undefined ? initialValue : array[0];
+  const startIndex = initialValue !== undefined ? 0 : 1;
```

Comment:
I really like your choice of `let/const`, good job! But:

It may sound strange, but usually we try to avoid negative comparisons if we can, so it would be much better to rewrite this code to: 

```suggestion
  let accumulator = initialValue === undefined ? array[0] : initialValue ;
  const startIndex = initialValue === undefined ? 1 : 0;
```